lms/djangoapps/static_template_view/tests/test_views.py

```diff
@@ -26,6 +32,64 @@ def test_about(self):
         """
         self._test_view('about', 'text/html')
 
+    def test_about_with_site_configuration(self):
+        """
+        Test the about view with the header and content set in SiteConfiguration.
+        """
+        test_header = "Very Unique Test Header"
+        test_content = "Very Unique Test Content"
+        test_header_key = 'static_template_about_header'
+        test_content_key = 'static_template_about_content'
+        response = None
+        configuration = {test_header_key: test_header, test_content_key: test_content}
+        with with_site_configuration_context(configuration=configuration):
+            response = self.client.get(reverse("about"))
+        self.assertIn(test_header, response.content)
+        self.assertIn(test_content, response.content)
+
+    def test_about_with_site_configuration_and_html_in_header(self):
+        """
+        Test the about view with html in the header.
+        """
+        test_header = "<i>Very Unique Test Header</i>"
+        test_content = "Very Unique Test Content"
+        test_header_key = 'static_template_about_header'
+        test_content_key = 'static_template_about_content'
+        response = None
+        configuration = {test_header_key: test_header, test_content_key: test_content}
+        with with_site_configuration_context(configuration=configuration):
+            response = self.client.get(reverse("about"))
+        self.assertIn(test_content, response.content)
+
+    def test_about_with_site_configuration_and_html_in_content(self):
+        """
+        Test the about view with html in the content.
+        """
+        test_header = "Very Unique Test Header"
+        test_content = "<i>Very Unique Test Content</i>"
+        test_header_key = 'static_template_about_header'
+        test_content_key = 'static_template_about_content'
+        response = None
+        configuration = {test_header_key: test_header, test_content_key: test_content}
+        with with_site_configuration_context(configuration=configuration):
+            response = self.client.get(reverse("about"))
+        self.assertIn(test_content, response.content)
+
+    def test_about_with_site_configuration_and_html_in_content_is_not_escaped(self):
+        """
+        Test the about view with html in content, making sure it isn't escaped.
+        """
+        test_header = "Very Unique Test Header"
+        test_content = "<i>Very Unique Test Content</i>"
+        test_header_key = 'static_template_about_header'
+        test_content_key = 'static_template_about_content'
+        response = None
+        configuration = {test_header_key: test_header, test_content_key: test_content}
+        with with_site_configuration_context(configuration=configuration):
+            response = self.client.get(reverse("about"))
+        escaped_content = escape(test_content)
+        self.assertNotIn(str(escaped_content), response.content)
```

Comment:
This test method seems unnecessary: the previous test confirms that the HTML content is in the response verbatim, so why run another test just to check that the escaped content isn't in the response.  If you want to explicitly check this condition, just add the assertNotIn to test_about_with_site_configuration_and_html_in_content

Reply:
This is a good question. In hindsight I am not 100% sure why I separated them. The only thing I can think of is that the unit test as a single unit is testing two different things, but in practicality both couldn't be happening at the same time. (Value is escaped and the value isn't escaped).